pyiron_base/generic/datacontainer.py

```diff
@@ -811,8 +811,14 @@ def _from_hdf(self, hdf, version=None):
             def normalize_key(name):
                 # split a dataset/group name into the position in the list and
                 # the key
-                k, i = name.split("__index_", maxsplit=1)
-                i = int(i)
+                try:
+                    k, i = name.split("__index_", maxsplit=1)
+                    i = int(i)
+                except ValueError:
+                    raise ValueError(
+                        f"Could not parse item name {name} in HDF group {hdf.h5_path}. "
+                        "Was this group really written by a DataContainer?"
```

Comment:
An error practically asking the user where the error comes from sounds super weird to me. I'd rather say that this element cannot have been written by `DataContainer` because it doesn't have the format "NAME__index_NUMBER".

This being said, can't we have an option to make all elements `DataContainer`-compatible? I mean, the sounds like a horrible hack that shouldn't happen by default, but it might make sense when we want to convert older stuff so having such a function might be useful (although I don't have a strong feeling for this so you can also ignore me on this point)

Reply:
> An error practically asking the user where the error comes from sounds super weird to me. I'd rather say that this element cannot have been written by `DataContainer` because it doesn't have the format "NAME__index_NUMBER".

Fair, I'll reword it.  The motivation was that this error mostly comes up for people using `DataContainer` in new job classes and make small mistakes and which point they are greeted by a weird `ValueError` from deep within their `from_hdf` without any indication what is the root cause.  This happened a few times now and while I can directly see the issue, it's difficult without more experience, so I want to transfer some of that knowledge in this better error message.

> This being said, can't we have an option to make all elements `DataContainer`-compatible? I mean, the sounds like a horrible hack that shouldn't happen by default, but it might make sense when we want to convert older stuff so having such a function might be useful (although I don't have a strong feeling for this so you can also ignore me on this point)

I can see that being useful for converting old classes, but would prefer that to be outside `DataContainer.from_hdf` either in a classmethod constructor or a separate function.  Implementation should be pretty trivial though.